### ISSUE TITLE
Allow routing by foot along `public_transport=platform` or `railway=platform`

### ISSUE THREAD

**Issue by joshinils** [valhalla/valhalla#4578]
Much like #3691 I'd like to suggest including any public transport platform for foot routing.
Yes, they can technically be closed to the public, but I've never seen that happen in Berlin unless the station itself is out of order, under repair etc.

Examples:
https://www.openstreetmap.org/directions?engine=fossgis_valhalla_foot&route=52.59788%2C13.33276%3B52.59589%2C13.33639
https://www.openstreetmap.org/directions?engine=fossgis_valhalla_foot&route=52.53626%2C13.34430%3B52.53613%2C13.34314
https://www.openstreetmap.org/directions?engine=fossgis_valhalla_foot&route=52.60508%2C13.32340%3B52.60732%2C13.32093

I understand including stairs may not be the best option and potentially lead to longer travel times.

This includes subway platforms, they could make a route underneath rivers or big road crossing.

**Comment by nilsnolde** [comment]
yeah, I was mentioning that too in the issue you're referring to. will have to refresh my mind how that interplays with GTFS/multimodal though..

**Comment by kevinkreiser** [comment]
when we hook up gtfs we look for edges to connect that have pedestrian access and have a "valid" use as defined here: 

https://github.com/valhalla/valhalla/blob/efff65d35c68c9198d504e07087d3ae2b71e5e1f/src/mjolnir/transitbuilder.cc#L34-L37

note that `kPlatform` is not one of them. so we will not connect in/egresses to these (which is good) the question is what to do with them regarding the connections from the stations to the transit lines (ie `kPlatformConnection` edges). ideally they would replace those but since they are modeled (maybe) in gtfs and then also modeled (maybe) in osm there would have to be a step to reconcile the two and frankly i think that is out of scope for valhalla. at most, i think if there is gtfs coverage in an area that overlaps an area that has osm platform edges, then we should throw out the osm and use whatever gtfs has. just my opinion though

**Comment by hungerburg** [comment]
Came here from this https://valhalla.openstreetmap.de/directions?profile=pedestrian&wps=11.327430009841919%2C47.26944549602832%2C11.329329013824465%2C47.26982041818458 - Other sample - planning trip from train station - https://valhalla.openstreetmap.de/directions?profile=pedestrian&wps=11.249694228172302%2C47.28096858849457%2C11.25378727912903%2C47.28018250654129 - In my opinion both these very `railway=platform` might merit an additional `highway=footway` tag; This combination is rare in data though, only 2.4% according to [taginfo](https://taginfo.openstreetmap.org/tags/railway%3Dplatform#combinations).

**Comment by joshinils** [comment]
> In my opinion both these very `railway=platform` might merit an additional `highway=footway` tag

Maybe in this case, but usually not all.

Especially for the usual S-Bahn stations I see in Berlin it is not a good idea, or other train stations from DB.

I don't know what kind of trains go to what you've linked, but from a quick glance at the aerial imagery on bing it should be possible to map that as an area, which be more precise, but in turn disallow tagging it as a footway, since it is not.

**Comment by matthiasfeist** [comment]
I found this issue recently as well while trying to route using the openstreetmap website:
https://www.openstreetmap.org/directions?engine=fossgis_valhalla_foot&route=59.319755%2C18.072097%3B59.318357%2C18.081908#map=15/59.31762/18.08375

Graphhopper correctly lets me walk along the bus platform, Valhalla doesn't and routes a pretty roundabout way because of the platform being a part of the walkway.

**Comment by eithanwes** [comment]
I was trying to route from train platforms and the route skips and starts from a nearby footway and not the platform itself.

https://www.openstreetmap.org/directions?engine=fossgis_valhalla_foot&route=48.205041%2C16.38385%3B48.20522%2C16.387154#map=18/48.205417/16.385754

in this case it is tagged `railway=platform_edge` but i think is applicable to all variants of public transit stops and platforms.

**Comment by nilsnolde** [comment]
Oh boy, another tag popping up every now and then.. yeah, I guess we should widen the current platform concept to all those. 

Would appreciate if someone here at least provide a comprehensive list of those ways and how they're tagged. That increases the chance dramatically that we'll take care of it.

**Comment by hungerburg** [comment]
After reading https://wiki.openstreetmap.org/wiki/Tag:railway=platform_edge I wager to say: This tag is not something, routers actually have to care about; At least not as a routable way. It always decorates a `*=platform` mapped as an area -- either walk along the edges (sic) of that or do plaza routing inside.

**Comment by eithanwes** [comment]
`highway=platform`
`railway=platform`
`public_transport=platform`
`railway=platform_edge`

those are the ones I found, others can add if they find others

**Comment by lumikeiju** [comment]
`public_transport=platform_edge` as well, though I will echo @hungerburg that it's not generally intended to itself be routable.
Usage count is `551` as of `2026-08-31` per [Taginfo](https://taginfo.openstreetmap.org/tags/public_transport=platform_edge#overview).
<img width="796" height="226" alt="Image" src="https://github.com/user-attachments/assets/e0f927c6-6072-4982-a334-3b5680fcb2d8" />
(I'm practically the only one who has used it in the US so far.)
Visualization via [QLever petrimaps](https://qlever.dev/petrimaps/?query=PREFIX+geo%3A+%3Chttp%3A%2F%2Fwww.opengis.net%2Font%2Fgeosparql%23%3E+PREFIX+osmkey%3A+%3Chttps%3A%2F%2Fwww.openstreetmap.org%2Fwiki%2FKey%3A%3E+SELECT+*+WHERE+%7B+%3Fosm+osmkey%3Apublic_transport+%22platform_edge%22+.+%3Fosm+geo%3AhasGeometry%2Fgeo%3AasWKT+%3Fgeometry+%7D&backend=https%3A%2F%2Fqlever.dev%2Fapi%2Fosm-planet).
